process_report/process_report.py

```diff
@@ -62,16 +64,42 @@ def is_old_pi(old_pi_dict, pi, invoice_month):
     return False
 
 
+def get_invoice_bucket():
+    try:
+        s3_resource = boto3.resource(
+            service_name="s3",
+            endpoint_url=os.environ.get(
+                "S3_ENDPOINT", "https://s3.us-east-005.backblazeb2.com"
+            ),
+            aws_access_key_id=os.environ["S3_KEY_ID"],
+            aws_secret_access_key=os.environ["S3_APP_KEY"],
+        )
+    except KeyError:
+        print("Error: Environment variables for S3 authentication not set")
+    return s3_resource.Bucket(os.environ["S3_BUCKET_NAME"])
```

Comment:
If S3_BUCKET_NAME isn't set it'll raise a KeyError which isn't handled. But, for our application we know the bucket name is `nerc-invoicing` so maybe set that as the default if it's not set?

Reply:
the bucket name should have been argument to this method, and the caller could look at the env variable and pass it to get the Bucket, but it's okay for now.

Reply:
Is it usually safe to provide a default value for the bucket name? What if the user intended to use a different bucket, and forgot to set this env var before running the script?

Reply:
The credentials should only work with a specific bucket unless the user decided to use credentials that can access all buckets then it's on the user.